lib/util/platform.js

```diff
@@ -354,6 +355,15 @@ shaka.util.Platform = class {
     return shaka.util.Platform.userAgentContains_('Sky_STB');
   }
 
+  /**
+   * Check if the current platform is XOne.
+   */
+  static isXOne() {
+    return navigator.platform &&
+        navigator.platform.includes('Linux aarch64') &&
+        shaka.util.Platform.userAgentContains_('AppleWebKit');
```

Comment:
Surely there is some more specific attribute you can check.  Maybe something outside the UA string, if the UA looks very generic?  I have to imagine any WebKit build on Arm Linux would be detected as XOne with this check.

Reply:
Agree but today i checked 


    navigator.platform
    < "Linux aarch64"

    navigator.userAgent
    < "Mozilla/5.0 (Linux; x86_64 GNU/Linux) AppleWebKit/601.1 (KHTML, like Gecko) Version/8.0 Safari/601.1 WPE"

    navigator.vendor
    < "Apple Computer, Inc."

Any other idea ?

Reply:
Maybe you can include `Version/8.0`

Reply:
ok

Reply:
Does this device expose any unusual properties anywhere within the global namespace?  Any API specific to this device?

Reply:
Let me check i will stringify all ;)

Reply:
I found that we have the RDK as a global variable. I used it too